### PR TITLE
added shellcheck directives

### PR DESCRIPTION
**Pull request by erouvas**
#### What is this PR About?
Addressed some issues raised by shellcheck and added directives to fine tune its opearation

#### How do we test this?
Functionality should have remained the same

cc: @redhat-cop/businessautomation-cop

### CODE REVIEW

File: bcgithook/install.sh [hunk 1]
```diff
@@ -52,6 +52,11 @@ bash_ok=no && [ "${BASH_VERSINFO:-0}" -ge $min_bash_version ] && bash_ok=yes
 # - declare some useful functions
 #
 
+# - shellcheck directives
```
Comment (garethahealy) [inline bcgithook/install.sh:55]: what are these comments for? are you expecting shell check to understand these and disable these checks?

File: bcgithook/scripts/post-commit.sh [hunk 2]
```diff
@@ -119,6 +119,7 @@ debug "START"
 [[ ! -r "$CONFIG_FILE" ]] && LOG_FILE="$ERR_FILE" \
                           && debug "CONFIG FILE $CONFIG_FILE NOT FOUND - ABORTING" \
                           && exit 1
+# shellcheck source=/dev/null
```
Comment (garethahealy) [inline bcgithook/scripts/post-commit.sh:122]: same as first comment. what are you expecting to happen

Comment (erouvas) [reply bcgithook/scripts/post-commit.sh:122]: yep, as per shellcheck documentation.

For example, check bottom of
https://github.com/koalaman/shellcheck/wiki/SC2034

File: bcgithook/install.sh (hunk 1 above)
Comment (erouvas) [reply bcgithook/install.sh:55]: pls see reply to comment below

Comment (garethahealy) [reply bcgithook/install.sh:55]: I've never done it that way, I've always passed as args using `-e`, like:
- https://github.com/koalaman/shellcheck/wiki/Ignore#ignoring-errors-in-one-specific-run
- https://github.com/marketplace/actions/shellcheck#globally-disable-checks

looking at the CI, not sure its worked either. still complains about SC2034

Comment (erouvas) [reply bcgithook/install.sh:55]: Worth trying with "-e" as well.

**Pull request merged**